File: FE/src/types/sticker.ts
export interface Sticker {
  id: number;
  name: string;
  imageUrl: string;
  category: string;
  tags: string;
  isPremium: boolean;
  premiumPrice: number;
  isActive: boolean;
  usageCount: number;
  createdAt: string;
}

export interface StickerCategory {
  category: string;
  count: number;
  stickers: Sticker[];
}

export interface Font {
  id: number;
  name: string;
  fontFamily: string;
  fontFileUrl: string;
  category: string;
  isPremium: boolean;
  premiumPrice: number;
  isActive: boolean;
  usageCount: number;
  previewText: string;
  createdAt: string;
}

File: BE/CleanArchitecture.Domain/Entities/Sticker.cs
using CleanArchitecture.Domain.Common;

namespace CleanArchitecture.Domain.Entities;

public class Sticker : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string ImageUrl { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public string Tags { get; set; } = string.Empty; // Comma-separated tags for search
    public bool IsPremium { get; set; } = false;
    public decimal PremiumPrice { get; set; } = 0;
    public bool IsActive { get; set; } = true;
    public int UsageCount { get; set; } = 0;
}
